src/index.js

```diff
@@ -184,7 +195,19 @@ exports.createFromBytes = (buf) => {
 }
 
 exports.createFromB58String = (str) => {
-  return new PeerIdWithIs(mh.fromB58String(str))
+  return exports.createFromCID(str) // B58String is CIDv0
+}
+
+exports.createFromCID = (cid) => {
+  if (typeof cid === 'string' || Buffer.isBuffer(cid)) {
+    cid = new CID(cid)
+  } else if (CID.isCID(cid)) {
+    CID.validateCID(cid) // throws on error
+  } else {
+    // provide more meaningful error than the one in CID.validateCID
+    throw new Error('Supplied cid value is neither String|CID|Buffer')
+  }
+  return new PeerIdWithIs(cid.multihash)
 }
```

Comment:
I think this could be as simple as:

```js
exports.createFromCID = cid => {
  cid = CID.isCID(cid) ? cid : new CID(cid)
  if (cid.version === 1 && cid.codec !== 'libp2p-key') throw new Error('invalid peer ID')
  return new PeerIdWithIs(cid.multihash)
}
```

Reply:
Yeah I like this more :raised_hands: